tfhe/src/integer/server_key/radix_parallel/sub.rs

```diff
@@ -407,6 +408,303 @@ impl ServerKey {
+        last_block_input_carry: &Ciphertext,
+        last_block_output_carry: &Ciphertext,
+    ) -> Ciphertext {
+        let bits_of_message = self.key.message_modulus.0.ilog2();
+
+        let resolve_overflow_lut = self.key.generate_lookup_table(|x| {
+            let carry_propagation = x >> bits_of_message;
+            let output_carry_of_block = (x >> 1) & 1;
+            let input_carry_of_block = x & 1;
+
+            // Resolve the carry that the last bit actually receives as input
+            let input_carry_to_last_bit = if carry_propagation == OutputCarry::Propagated as u64 {
+                input_carry_of_block
+            } else if carry_propagation == OutputCarry::Generated as u64 {
+                1
+            } else {
+                0
+            };
+
+            u64::from(input_carry_to_last_bit != output_carry_of_block)
+        });
+
+        let x = self.key.unchecked_scalar_mul(last_block_output_carry, 2);
+        self.key
+            .unchecked_add_assign(&mut last_block_inner_propagation, &x);
+        self.key
+            .unchecked_add_assign(&mut last_block_inner_propagation, last_block_input_carry);
+        self.key
+            .apply_lookup_table(&last_block_inner_propagation, &resolve_overflow_lut)
+    }
+
+    // This is the implementation of overflowing sub when we can use parallel carry
+    // propagation.
+    //
+    // It is in its own function so that it can be tested, as the main entry point
+    // unchecked_signed_overflowing_sub may select non parallel version if lhs
+    // does not have enough block.
+    pub(crate) fn unchecked_signed_overflowing_sub_parallelized_impl(
+        &self,
+        lhs: &SignedRadixCiphertext,
+        rhs: &SignedRadixCiphertext,
+    ) -> (SignedRadixCiphertext, Ciphertext) {
+        // This assert is here because this overflow computation requires these preconditions
+        // which is_eligible_for_parallel_single_carry_propagation, but it could change in the
+        // future
+        assert!(self.key.message_modulus.0 >= 4 && self.key.carry_modulus.0 >= 4);
+
+        // In Two's complement arithmetic, overflow occurs when the output carry of the
+        // last bit is not the same as the input carry of the last bit.
+        //
+        // Here we have blocks, and we cannot just compare input and output carries of the last
+        // block as its not equivalent to checking what happens on the last bit.
+        // So we have to resolve that carry propagation that happens in the last block.
+        //
+        // So the carry propagation is done in 2 steps, first we compute the carry propagation
+        // in the last block to be able at the second step, to know the actual carry that
+        // the last bit receives.
+        //
+        // These are done in parallel to other stuff, and so no additional 'latency cost'
+        // should occur.
+
+        let mut result = lhs.clone();
+
+        let neg = self.unchecked_neg(rhs);
```

Comment:
It does not exists, I also don't think it can be made parallelized, and from measures of `unchecked_add_parallelized` it  would not bring measurable improbements

Reply:
ok !

Reply:
hum does not seems a good thing, I'm worried ppl will say `hey why is unchecked_neg_parallelized not faster than unchecked_neg ?`